precompile/allow_list.go

```diff
@@ -59,55 +61,45 @@ func (c *AllowListConfig) Equal(other *AllowListConfig) bool {
 	if other == nil {
 		return false
 	}
-	if len(c.AllowListAdmins) != len(other.AllowListAdmins) {
+	if !areEqualAddressLists(c.AllowListAdmins, other.AllowListAdmins) {
+		return false
+	}
+
+	return areEqualAddressLists(c.EnabledAddresses, other.EnabledAddresses)
```

Comment:
doesn't this make it impossible to change the order of addresses?

Reply:
yes. do you think we should flex this?

Reply:
We could also add a function to the interface that returns a canonical representation that can be used in hashing.

Reply:
I don't think we need to sort these lists. I think it's ok to expect the same order every time since `hash(list[b,c,a]) != hash(list[a,b,c])`. This is because the byte representation of those lists are not equal. JSON arrays are also ordered so users should intuitively know that providing any different list can cause issues.

Reply:
Since `Equal` will be used by the node locally to determine the validity of an upgrade config, if we change this in the future to be "looser" or to sort these lists, then the upgrade process should still be straight forward ie.

we push an upgrade that changes the `Equal` check to to define equality to ensure that the sorted list is equal and an upgrade that would have previously been invalid (ie a list in a different order, but containing the same set of addresses) will become valid, which is fine since it's just for the node's upgrade config.

We should also make sure that we do not allow duplicates in the list. Created a ticket for this here, since I think this PR is ready to merge: https://github.com/ava-labs/subnet-evm/issues/269.